test/test13.erl

```diff
@@ -0,0 +1,8 @@
+-module(test13).
+-export([f/0]).
+
+%% test case for list expressions
+f() ->
+    List = [3, 1, 4, 1, 5],
+    List = [3 | [1 | [4 | [1 | [5 | []]]]]],
+    [X * 2 || X <- List, X >= 3].
```

Comment:
Could you add an improper list (like `[1 | 2]`) pattern and a list like `[1, 2, 3 | [4]]`?

Reply:
I added some test cases.